python/lsst/meas/algorithms/astrometrySourceSelector.py

```diff
@@ -120,14 +122,22 @@ def _isMultiple(self, source):
 
     def _hasCentroid_vector(self, sourceCat):
         """Return True for each source that has a valid centroid"""
-        return np.isfinite(sourceCat.get(self.centroidXKey)) \
-            & np.isfinite(sourceCat.get(self.centroidYKey)) \
+        def centroid_nonfinite_but_not_flagged():
+            """Return True for sources with non-flagged, non-finite centroids."""
+            return ((~np.isfinite(sourceCat.get(self.centroidXKey)) |
+                    ~np.isfinite(sourceCat.get(self.centroidYKey))) &
+                    ~sourceCat.get(self.centroidFlagKey))
+        assert ~centroid_nonfinite_but_not_flagged().any(), \
+            "Centroids not finite for %d unflagged sources." % (centroid_nonfinite_but_not_flagged().sum())
```

Comment:
You can actually just do
```
assert np.isfinite(sourceCat.get(self.centroidXKey)).all() and np.isfinite(sourceCat.get(self.centroidYKey)).all()
```
In other words, centroid values should never be NaN, *even when the flag is set*.  And that's why I wanted to make that an assert: if a centroid algorithm disobeys that rule, it's a serious bug.

Reply:
Also, I'm obligated to remind you that snake case isn't allowed :(